latest changes

mm1ke opened the pull request:
This changes adds:
- tapdel: for deleting unused qtap devices
- save: for saving changes to a harddisk if vm starts in snapshotmode
- debug flag: output the vm settings and full start command
- updated 'network code': its possible to start vm without a network card and it should be easier now to add new network settings. Now you can set "bridge" and "none" as network-typ.

mm1ke committed:
add DEBUG flag which outputs the full start command

mm1ke committed:
fixed debug output, didn't check for the debug flag

mm1ke committed:
minor spelling changes

mm1ke committed:
made bridge optional and make it possible to start guest without any network, it's also easier to add other network types

mm1ke committed:
add discription, add tapdel and save options

mm1ke committed:
nicer debug output

mm1ke committed:
update debug output

mm1ke committed:
if statement didn't work with '[[' and ']]'

jlec commented on kvm.init:
```diff
@@ -42,6 +48,9 @@ PID_DIR="/var/run/kvm"
 TMP_DIR="/var/tmp"
 
+# 1=on, 0=off
+DEBUG=0
```
Would it be better to use string type values like "yes" or "true"? This would make the check later look nicer.

jlec commented on kvm.init:
```diff
@@ -67,6 +76,28 @@ scan_running_vm(){
+# tapdel()
+# removes unused tap devices
+#
+tapdel(){
```
You could add

local active_qtab

jlec commented:
We should agree on an indention type. You are indenting with 2 spaces where I am indenting with tab. Personally I don't care what we are doing, but we should keep it consistent.

mm1ke committed:
typo

mm1ke committed:
debug: using true/false instead of 1/0

mm1ke committed:
add local vars

mm1ke committed:
tapdel didn't work

mm1ke committed:
really start guest without a network card if VM_NET_TYP is set to none

mm1ke committed:
add usermode network

mm1ke committed:
rename bridge network to tap, qemu has also an bridge network mode which seems to be different from tap mode. see qemu-system-x86_64 --help

mm1ke committed:
small case fix

mm1ke commented:
Hi Justin. I've changed Debug to true/false and added local vars. It's also clearly a good idea to keep indenting consistent. Actually i'm wondering what editor you use. I'm using kate with the default indenting settings, which means a tab is 2 spaces for me (usually)...
Anyway, i see my changes are now in this pull request too (didn't know that i added those changes to this pull request..). As you can see there are a few other changes/fixes too.

jlec commented:
I am using nano, so every indention is manual.
So should we agree in 2 spaces then?

mm1ke commented:
Yeah, it would be great if we could stay with 2 spaces :)

mm1ke committed:
make daemonize optional

mm1ke committed:
make smp/memory settings optional, qemu defaults to 128mb memory and smp=1 anyway

mm1ke committed:
add possibilty to set machine typ of an guest

mm1ke committed:
add possiblity to start from a kernel + initrd file

mm1ke committed:
fix when testing VM_KERNEL & VM_APPEND

mm1ke committed:
fix VM_APPEND again

mm1ke committed:
fix VM_NET_TYP check

mm1ke committed:
add arch specific configuration, more debug output

mm1ke committed:
fix save command

mm1ke committed:
add warning for save command

mm1ke committed:
typo

mm1ke committed:
typo

mm1ke committed:
mips specific changes

mm1ke committed:
fix: check for  in tapdel() if script ist called directly

mm1ke committed:
fix if statement

mm1ke committed:
set VM_NET_TYP to none at stop() if nothing is set

mm1ke committed:
put VM_NAME out of the functions and set it globally

mm1ke committed:
make myconfig and KVM_PID a local var

mm1ke committed:
myconfig can't be local, it's used in start()

mm1ke committed:
typo

mm1ke committed:
cosmetic changes

mm1ke committed:
no need to check if file exists, rm --force ignores non existing files

mm1ke committed:
fix: mkdir command wasn't declared

mm1ke committed:
rewrite stop function, rm -rf dosen't need a check if file exists

mm1ke committed:
minor changes (comments)

mm1ke committed:
simpler stop_tap_device

mm1ke committed:
keep code consistent

mm1ke committed:
check for max vms and make it possible to change the value

mm1ke committed:
add comment

mm1ke committed:
fix: correct check for VM_KERNEL and VM_INITRD - VM_BOOT_DEV would set to nothing everytime

mm1ke committed:
add comment about the non wroking commit feature, also add link to bugreport

mm1ke committed:
fix: missing then in if statement

mm1ke committed:
default.config: rearrange config a bit

mm1ke committed:
kvm.init: new switch snapshot: start a vm in snapshot (overwrites config)

mm1ke committed:
Revert "kvm.init: new switch snapshot: start a vm in snapshot (overwrites config)"

This reverts commit b9557da0c6f5b60804ab7dddd1c6a8046f091dfa.

mm1ke committed:
get rid of grep and cut, BRDEV has to be set in the default configuration file so that tapdel() works with kvm.init too

mm1ke committed:
remove on extra assignment

mm1ke committed:
remove save from extra_commands aslong it dosn't work

mm1ke committed:
simpler checks

mm1ke committed:
cfg-check shouldn't warn when vm_net_typ is set to none

mm1ke committed:
implement info() - shows information about running vms

mm1ke committed:
info() bug fixes

mm1ke committed:
update info(), nicer output, fix bugs)

mm1ke committed:
using -o command= instead of -p -f --no-heading

mm1ke committed:
don't start vm as root if user not found. arbort start

mm1ke committed:
update error/warn/info messages

mm1ke committed:
update output

mm1ke committed:
remove save() from extra_commands aslong it doesn't work

mm1ke committed:
merge changes

mm1ke committed:
add missing changes again

mm1ke committed:
also print info about snapshotmode

mm1ke committed:
add missing local

mm1ke committed:
indentation fixes

mm1ke committed:
cleanups

mm1ke committed:
get the information with grep regex pattern, remove cut again, remove einfon/echo workaround

mm1ke committed:
really remove cut

mm1ke committed:
fix: add missing echo

mm1ke committed:
implement kvm.init info

mm1ke committed:
for address_port we need head -n1 since there could be more than one address

mm1ke committed:
add workaround for gentoo bug 461744

mm1ke committed:
ip moved from sbin to bin

mm1ke committed:
remove TODO/obsolet comment

mm1ke committed:
update comment

mm1ke committed:
remove vars for programs which are part of the system

mm1ke committed:
typo

mm1ke committed:
rename a few VARS and add VM_ENABLE_KVM

mm1ke committed:
add VM_ENABLE_KVM Var

mm1ke committed:
rename VARs to suit with default.config

mm1ke committed:
using a  for the configuration path

mm1ke committed:
add warning regarding cpu=host

mm1ke committed:
remove x from kvm.init
